compiler/src/main/scala/org/scalajs/core/compiler/GenJSExports.scala

```diff
@@ -124,9 +124,9 @@ trait GenJSExports extends SubComponent { self: GenJSCode =>
       for {
         m <- genTopLevelOrStaticExports(classSym, ExportDestination.TopLevel)
       } yield {
-        m match {
+        (m: @unchecked) match {
           case m: js.TopLevelFieldExportDef => m
-          case _                            => js.TopLevelExportDef(m)(m.pos)
+          case m: js.MethodDef              => js.TopLevelMethodExportDef(m)(m.pos)
```

Comment:
Consider pushing this inside `genTopLevelOrStaticMethodExports`. It would avoid this weird mismatch altogether (and in there it seems completely acceptable, if not for the parallel with `genTopLevelOrStaticFieldExports`).